### ISSUE TITLE
kdump-tools_1.6.1-1.orig.tar.gz not present and build failing

### ISSUE THREAD

**Issue by ssudarshCLS** [sonic-net/sonic-buildimage#14929]
#### Steps to reproduce the issue:
1. Cloned the 202106 branch
2. make init
3. NOSTRETCH=1 NOBUSTER=0 NOBULLSEYE=1 PLATFORM=broadcom configure
4. NOSTRETCH=1 NOBUSTER=0 NOBULLSEYE=1 SONIC_BUILD_JOBS=4 make target/sonic-broadcom.bin
#### Describe the results you received:

--2023-05-03 14:05:19--  http://deb.debian.org/debian/pool/main/m/makedumpfile/makedumpfile_1.6.1.orig.tar.gz
Resolving deb.debian.org (deb.debian.org)... 199.232.46.132, 2a04:4e42:48::644
Connecting to deb.debian.org (deb.debian.org)|199.232.46.132|:80... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-05-03 14:05:19 ERROR 404: Not Found.

make[1]: *** [Makefile:8: /sonic/target/debs/buster/kdump-tools_1.6.1-1_all.deb] Error 8
make[1]: Leaving directory '/sonic/src/kdump-tools'
[  FAIL LOG END  ] [ target/debs/buster/kdump-tools_1.6.1-1_all.deb ]
make: *** [slave.mk:435: target/debs/buster/kdump-tools_1.6.1-1_all.deb] Error 1

#### Output of `show version`:

```
(paste your output here)
```
#### Output of `show techsupport`:

```
(paste your output here or download and attach the file here )
```

**Comment by ssudarshCLS** [comment]
![image](https://user-images.githubusercontent.com/127869045/235947070-d4b86766-79f6-421a-9ae6-f2b129d36997.png)

Seems like some changes have been made in the URL http://deb.debian.org/debian/pool/main/m/ that would have removed the 1.6.1 version.

But am surprised as to how the community Azure Pipeline build succeeded.

https://dev.azure.com/mssonic/build/_build/results?buildId=265618&view=logs&j=88ce9a53-729c-5fa9-7b6e-3d98f2488e3f&t=b71397ab-4d88-53c1-4db7-b4fc5ec36f6f

**Comment by gechiang** [comment]
@liushilongbuaa can you check if the cache some how may have masked off this issue that azure pipeline did not catch?

**Comment by liushilongbuaa** [comment]
#curl http://deb.debian.org/debian/pool/main/m/makedumpfile/makedumpfile_1.6.1.orig.tar.gz -o makedumpfile_1.6.1.orig.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  172k  100  172k    0     0  12.9M      0 --:--:-- --:--:-- --:--:-- 12.9M
# ls makedumpfile_1.6.1.orig.tar.gz -al
-rw-r--r-- 1 root root 176228 May 11 02:04 makedumpfile_1.6.1.orig.tar.gz

**Comment by liushilongbuaa** [comment]
In my VM(azure westus3). That url still exists.
So it seems related with CDN?

**Comment by samaity** [comment]
@gechiang  is anybody working on it?

**Comment by liushilongbuaa** [comment]
community Azure Pipeline build succeeded because of we enabled reproducible build feature.
Options: 'SONIC_VERSION_CONTROL_COMPONENTS=deb,py2,py3,web,git,docker MIRROR_SNAPSHOT=y'

We have a debian mirror and webfile storage. That file is dumped by wget. It stores in Azure storage.

**Comment by liushilongbuaa** [comment]
@samaity , @ssudarshCLS , community build enabled reproducible build.
SONIC_VERSION_CONTROL_COMPONENTS=deb,py2,py3,web,git,docker

It belongs to 'web'.

**Comment by liushilongbuaa** [comment]
We dumped web files to Azure storage.

**Comment by ssudarshCLS** [comment]
@liushilongbuaa , could you kindly elaborate a bit more on the reproducible build?
Also I am building directly from the source without any intermediate storage/caching for web files.
So my expectation of successful compilation from source is correct?

**Comment by liushilongbuaa** [comment]
> @liushilongbuaa , could you kindly elaborate a bit more on the reproducible build? Also I am building directly from the source without any intermediate storage/caching for web files. So my expectation of successful compilation from source is correct?

Updating kdump-tools to available version is correct. Building from source is correct.

But to unblock others, we used reproducible build options in PR checks and official build.

NOSTRETCH=1 NOBUSTER=0 NOBULLSEYE=1 SONIC_BUILD_JOBS=4 make target/sonic-broadcom.bin SONIC_VERSION_CONTROL_COMPONENTS=deb,py2,py3,web,git,docker MIRROR_SNAPSHOT=y

**Comment by liushilongbuaa** [comment]
This is a part of build improvement.
https://github.com/sonic-net/SONiC/blob/master/doc/sonic-build-system/SONiC-Reproduceable-Build.md

**Comment by ssudarshCLS** [comment]
@liushilongbuaa 
I tried the Reproduceable build as suggested by you and didn't encounter kdump-tools target build failure.
But I faced some weird failure,

![image](https://github.com/sonic-net/sonic-buildimage/assets/127869045/9c655339-5757-4986-8a96-465bf834b6d8)

Have you faced the above?

Also I tried looking for kdump-tools web package URL inside /work/users/ssudarsh/sonic-net-202106/files/build/versions/default for the modified URL of kdump-tools for reproduceable build but couldn't find it.
So is kdump-tools taken care by reproduceable build?

**Comment by ssudarshCLS** [comment]
The error is coming because of the class `UnixHTTPConnection` inside /usr/local/lib/python3.7/dist-packages/docker/transport/unixconn.py part of the docker

```python3
import requests.adapters
import socket
import http.client as httplib

from docker.transport.basehttpadapter import BaseHTTPAdapter
from .. import constants

try:
    import requests.packages.urllib3 as urllib3
except ImportError:
    import urllib3


RecentlyUsedContainer = urllib3._collections.RecentlyUsedContainer


class UnixHTTPConnection(httplib.HTTPConnection):
```

```bash
(Pdb) inspect.getfullargspec(httplib.HTTPConnection.request)
FullArgSpec(args=['self', 'method', 'url', 'body', 'headers'], varargs=None, varkw=None, defaults=(None, {}), kwonlyargs=['encode_chunked'], kwonlydefaults={'encode_chunked': False}, annotations={})

```

The httplib.HTTPConnection.request function that gets inherited doesn't support the chunked kwarg.

The version of docker python module being installed inside the fsroot-broadcom  which is chrooted is **5.0.3**

![image](https://github.com/sonic-net/sonic-buildimage/assets/127869045/0c991e34-6948-4798-86cd-386e07c6dab1)


I know this is an altogether new issue. But would like to know that inspite of using reproduceable builds am encountering this error.

**Comment by liushilongbuaa** [comment]
This is caused by urllib3V2. Do you enabled py2,py3 options?
docker==5.0.3 requires urllib3<2.

**Comment by liushilongbuaa** [comment]
I have a draft PR https://github.com/sonic-net/sonic-buildimage/pull/15181
But it has other issues. Still checking.

**Comment by liushilongbuaa** [comment]
https://github.com/sonic-net/sonic-buildimage/blob/de3b9d24a9d9eb0eafc9a3b1d5510f82556dd08e/build_debian.sh#L439
It is updated to 6.1.1 recently.

**Comment by srthati** [comment]
@liushilongbuaa, is Reproduceable build not supported for arm64? I'm getting below error with SONIC_VERSION_CONTROL_COMPONENTS set to all.
Build Configuration
"CONFIGURED_PLATFORM"             : "marvell-arm64"
"CONFIGURED_ARCH"                 : "arm64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "48"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"SONIC_BUFFER_MODEL"              : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : "http://72.163.217.104:80"
"HTTPS_PROXY"                     : "http://72.163.217.104:80"
"NO_PROXY"                        : ".local,.cisco.com,localhost,.nip.io,10.95.1.1"
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20230523.142818"
"BUILD_LOG_TIMESTAMP"             : "none"
"SONIC_IMAGE_VERSION"             : "HEAD.0-dirty-20230523.142818"
"BLDENV"                          : "buster"
"VS_PREPARE_MEM"                  : "yes"
"INCLUDE_MGMT_FRAMEWORK"          : "n"
"INCLUDE_ICCPD"                   : "n"
"INCLUDE_SYSTEM_TELEMETRY"        : "y"
"ENABLE_HOST_SERVICE_ON_START"    : "n"
"INCLUDE_RESTAPI"                 : "n"
"INCLUDE_SFLOW"                   : "y"
"INCLUDE_NAT"                     : "y"
"INCLUDE_DHCP_RELAY"              : "y"
"INCLUDE_KUBERNETES"              : "n"
"INCLUDE_MACSEC"                  : "y"
"TELEMETRY_WRITABLE"              : ""
"PDDF_SUPPORT"                    : "y"
"MULTIARCH_QEMU_ENVIRON"          : "y"
"SONIC_VERSION_CONTROL_COMPONENTS": "all"

fatal: not a git repository (or any of the parent directories): .git
"SONIC_DPKG_CACHE_METHOD"         : "none"

[ FAIL LOG START ] [ target/debs/buster/linux-image-4.19.0-12-arm64.deb ]
[ REASON ] :      NEWER PREREQUISITES: .platform  
[ FLAGS  FILE    ] : [] 
[ FLAGS  DEPENDS ] : [] 
[ FLAGS  DIFF    ] : [] 
Failed to verify the package: https://github.com/Marvell-switching/sonic-marvell-binaries/raw/master/arm64/kernel/linux-image-4.19.0-12-arm64.deb, the version is not specified
[  FAIL LOG END  ] [ target/debs/buster/linux-image-4.19.0-12-arm64.deb ]
make: *** [slave.mk:332: target/debs/buster/linux-image-4.19.0-12-arm64.deb] Error 1

**Comment by srthati** [comment]
> @srthati which branch?

202106

**Comment by srthati** [comment]
@liushilongbuaa 

I managed to update proxy setting could proceed further but hit another issue with below error

"E: Unable to execute target architecture"

sudo debootstrap --verbose --variant=minbase --arch $CONFIGURED_ARCH --unpack-tarball=$BASEIMAGE_TARBALL $IMAGE_DISTRO $FILESYSTEM_ROOT

above from scripts/build_debian_base_system.sh is failing.

is reproducible build supported for arm64? who can help on this?

build command:

time make http_proxy=http://72.163.217.104:80 https_proxy=http://72.163.217.104:80 target/sonic-marvell-arm64.bin NOSTRETCH=1 NOBUSTER=0 NOBULLSEYE=1 SONIC_VERSION_CONTROL_COMPONENTS=deb,py2,py3,web,git,docker MIRROR_SNAPSHOT=y

label:
commit de3b9d24a9d9eb0eafc9a3b1d5510f82556dd08e (HEAD -> 202106, origin/202106)


Thanks,
Srinivas

**Comment by liushilongbuaa** [comment]
We didn't setup pipeline to build marvell-arm64 in 202106 branch.
For marvell-arm64's build, we are trying to enable in master branch. I will try to reproduce the issue locally.

**Comment by jhli-cisco** [comment]
We hit the exact failure.

First we hit the makedumpfile issue, followed suggestion to add “SONIC_VERSION_CONTROL_COMPONENTS=all” and “MIRROR_SNAPSHOT=y” in the build cmd, got past the original build failure at “makedumpfile”, then hit the failure at generating the final .bin image:
 
+ echo '[INFO] Build host debian base system...'
[INFO] Build host debian base system...
+ TARGET_PATH=target
+ scripts/build_debian_base_system.sh amd64 buster ./fsroot
--2023-05-30 21:17:58--  http://packages.trafficmanager.net/snapshot/debian/20230509T000212Z/pool/main/a/adduser/adduser_3.118_all.deb
Resolving packages.trafficmanager.net (packages.trafficmanager.net)... 13.107.238.69, 13.107.237.69, 2620:1ec:bdf::69, ...
Connecting to packages.trafficmanager.net (packages.trafficmanager.net)|13.107.238.69|:80... failed: Connection timed out.
Connecting to packages.trafficmanager.net (packages.trafficmanager.net)|13.107.237.69|:80... failed: Connection timed out.
Connecting to packages.trafficmanager.net (packages.trafficmanager.net)|2620:1ec:bdf::69|:80... failed: Cannot assign requested address.
Connecting to packages.trafficmanager.net (packages.trafficmanager.net)|2620:1ec:46::69|:80... failed: Cannot assign requested address.
Retrying.
 
The failure is due to “SONIC_VERSION_CONTROL_COMPONENTS=deb” setting doesn’t work with build environment that requires http proxy. In https://github.com/sonic-net/sonic-buildimage/blob/202012/build_debian.sh#L80, build_debian_base_system.sh is invoked without 4th parameter, hence https://github.com/sonic-net/sonic-buildimage/blob/202012/scripts/build_debian_base_system.sh#L6 is not defined. If “SONIC_VERSION_CONTROL_COMPONENTS=deb” is set,  https://github.com/sonic-net/sonic-buildimage/blob/202012/scripts/build_debian_base_system.sh#L21 is skipped also which will result in http_proxy unset.
 
We did get successful build with SONIC_VERSION_CONTROL_COMPONENTS=py2,py3,web,git,docker.

**Comment by liushilongbuaa** [comment]
@jhli-cisco , could you please create PR to fix in master branch? Then add 'Request for xxx' branch tag.

**Comment by jhli-cisco** [comment]
> @jhli-cisco , could you please create PR to fix in master branch? Then add 'Request for xxx' branch tag.

A PR is raised at https://github.com/sonic-net/sonic-buildimage/pull/15268

**Comment by srthati** [comment]
Thanks @jhli-cisco

**Comment by srthati** [comment]
@jhli-cisco 

I tried new build with below PRs changes.

https://github.com/sonic-net/sonic-buildimage/pull/15181
https://github.com/sonic-net/sonic-buildimage/pull/15268

makedumpfile and proxy issues are resolved but still hitting failure  in "debootstrap"

error: "E: Unable to execute target architecture"

sudo debootstrap --verbose --variant=minbase --arch $CONFIGURED_ARCH --unpack-tarball=$BASEIMAGE_TARBALL $IMAGE_DISTRO $FILESYSTEM_ROOT

Thanks,
Srinivas

**Comment by liushilongbuaa** [comment]
@srthati , It seems PLATFORM_ARCH configure issue.
